### PR TITLE
[quant][graphmode][fx] Support preserving attributes in deepcopy of observed/quantized graphmodule

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56550 [quant][graphmode][fx] Support preserving attributes in deepcopy of observed/quantized graphmodule**

Summary:
Add support for preserving a list of attributes on observed/quantized GraphModule

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_deepcopy_preserve_attributes

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27899317](https://our.internmc.facebook.com/intern/diff/D27899317)

### CODE REVIEW

**Review by t10-13rocket — APPROVED** [review]
LGTM

**Review by t10-13rocket — APPROVED** [review]
LGTM

File: torch/quantization/quantize_fx.py [hunk 1]
```diff
@@ -233,6 +236,11 @@ def fuse_fx(model: torch.nn.Module,
     torch._C._log_api_usage_once("quantization_api.quantize_fx.fuse_fx")
     assert not model.training, 'fuse_fx only works on models in eval mode'
     graph_module = torch.fx.symbolic_trace(model)  # type: ignore
+    preservd_attributes: Set[str] = set()
```
Comment (vkuzo) [inline torch/quantization/quantize_fx.py:239]: should this be `preserved_attributes` (typo)?

Comment (vkuzo) [reply torch/quantization/quantize_fx.py:239]: can we update the docblock of `fuse_fx` and `prepare_fx` with this?

**Review by vkuzo — APPROVED** [review]

**Pull request closed without merging**